platforms/software/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/api/InvalidConfigurationUsageIntegrationTest.groovy

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.resolve.api
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+
+class InvalidConfigurationUsageIntegrationTest extends AbstractIntegrationSpec {
+    def "causing invalid resolution via #methodName for role #role fails"() {
+        given:
+        buildFile << """
+            import org.gradle.api.internal.artifacts.configurations.ConfigurationRole
+
+            configurations.$role('custom')
+            configurations.custom.$methodCall
+        """
+
+        when:
+        executer.noDeprecationChecks() // These will be checked elsewhere, this test is about ensuring failures
```

Comment:
🤔 What happens if we remove this? Are we getting any deprecations in this test? I wouldn't expect any